gmachine.go

```diff
@@ -172,52 +176,109 @@ func (g *GMachine) ExecuteBinary(binPath string) error {
 func Assemble(code []string) ([]Word, error) {
 	words := []Word{}
 	constants := PredefinedConstants
+	runes := Runes
 	for pos := 0; pos < len(code); pos++ {
 		token := code[pos]
-		var word Word
-
-		if strings.HasSuffix(token, ":") {
-			fmt.Println("start routine")
+		if token == "" {
+			continue
 		}
-
 		instruction, ok := TranslateTable[token]
-		if ok {
-			word = instruction.Opcode
-		} else {
-			value, err := strconv.Atoi(token)
-			if err != nil {
-				return nil, fmt.Errorf("invalid instruction %q at postion %d", token, pos)
+		if !ok {
+			//fmt.Println(token, "is data")
+			token = strings.ReplaceAll(token, "'", "")
+			word, ok := runes[token]
+			if !ok {
+				temp, err := strconv.Atoi(token)
+				if err != nil {
+					return nil, err
+				}
+				word = Word(temp)
 			}
-			word = Word(value)
-
+			words = append(words, word)
+			continue
 		}
-		words = append(words, word)
+		//fmt.Println(token, "is opcode")
+		words = append(words, instruction.Opcode)
 		if instruction.Operands > 0 {
```

Comment:
It's feeling like this is now getting so deeply nested it's worth moving out to a function, isn't it?

Reply:
sounds reasonable to me.

Reply:
started by `AssembleData` function